### ISSUE TITLE
Fails to build 

### ISSUE THREAD

**Issue by Lyamc** [koverstreet/bcachefs#312]
Ubuntu 21.04, gcc 10.3.0

```
$ make
  CALL    scripts/checksyscalls.sh
  CALL    scripts/atomic/check-atomics.sh
  DESCEND objtool
  CHK     include/generated/compile.h
  CC      fs/bcachefs/btree_iter.o
fs/bcachefs/btree_iter.c: In function ‘bch2_btree_iter_verify_locks’:
fs/bcachefs/btree_iter.c:392:49: error: parameter name omitted
  392 | static inline void bch2_btree_iter_verify_locks(struct btree_trans *, struct btree_iter *iter) {}
      |                                                 ^~~~~~~~~~~~~~~~~~~~
make[2]: *** [scripts/Makefile.build:272: fs/bcachefs/btree_iter.o] Error 1
make[1]: *** [scripts/Makefile.build:515: fs/bcachefs] Error 2
make: *** [Makefile:1847: fs] Error 2
```

```
$ gcc -v
Using built-in specs.
COLLECT_GCC=/usr/bin/gcc-10
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/10/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 10.3.0-1ubuntu1' --with-bugurl=file:///usr/share/doc/gcc-10/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-10 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-10-gDeRY6/gcc-10-10.3.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-10-gDeRY6/gcc-10-10.3.0/debian/tmp-gcn/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-mutex
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.3.0 (Ubuntu 10.3.0-1ubuntu1)
```

**Comment by Lyamc** [comment]
Switched to gcc-11 and it completed for some reason, ticket closed.

**Comment by koverstreet** [comment]
This was actually a real bug - fixed by f2b722c9bb
